solidity/ecdsa/test/WalletFactory.DKG.test.ts

```diff
@@ -0,0 +1,2769 @@
+            before(async () => {
+              await createSnapshot()
+
+              await mineBlocksTo(
+                resultSubmissionBlock + params.dkgResultChallengePeriodLength
+              )
+            })
+
+            after(async () => {
+              await restoreSnapshot()
+            })
+
+            context("when called by a DKG result submitter", async () => {
+              let tx: ContractTransaction
+              // let initialDkgRewardsPoolBalance: BigNumber
+              // let initialSubmitterBalance: BigNumber
+
+              before(async () => {
+                await createSnapshot()
+
+                // initialDkgRewardsPoolBalance =
+                //   await walletFactory.dkgRewardsPool()
+                // initialSubmitterBalance = await testToken.balanceOf(
+                //   await submitter.getAddress()
+                // )
+                tx = await walletFactory
+                  .connect(submitter)
+                  .approveDkgResult(dkgResult)
+              })
+
+              after(async () => {
+                await restoreSnapshot()
+              })
+
+              it("should emit DkgResultApproved event", async () => {
+                await expect(tx)
+                  .to.emit(walletFactory, "DkgResultApproved")
+                  .withArgs(dkgResultHash, await submitter.getAddress())
+              })
+
+              it("should clean dkg data", async () => {
+                await assertDkgResultCleanData(walletFactory)
+              })
+
+              it("should activate a candidate wallet", async () => {
+                expect(await wallet.activationBlockNumber()).to.be.equal(
+                  tx.blockNumber
+                )
+              })
+
+              // it("should reward the submitter with tokens from DKG rewards pool", async () => {
+              //   const currentDkgRewardsPoolBalance =
+              //     await walletFactory.dkgRewardsPool()
+              //   expect(
+              //     initialDkgRewardsPoolBalance.sub(currentDkgRewardsPoolBalance)
+              //   ).to.be.equal(params.dkgResultSubmissionReward)
+
+              //   const currentSubmitterBalance: BigNumber =
+              //     await testToken.balanceOf(await submitter.getAddress())
+              //   expect(
+              //     currentSubmitterBalance.sub(initialSubmitterBalance)
+              //   ).to.be.equal(params.dkgResultSubmissionReward)
+              // })
+
+              it("should emit WalletActivated event", async () => {
+                await expect(tx)
+                  .to.emit(walletFactory, "WalletActivated")
+                  .withArgs(wallet.address)
+              })
+
+              it("should unlock the sortition pool", async () => {
+                await expect(await sortitionPool.isLocked()).to.be.false
+              })
+
+              it("should transfer wallet ownership to wallet registry")
```

Comment:
Are there any plans to implement this one? If yes, can we add a todo? We have in total 3 "its" like this in this file.